sql/core/src/test/scala/org/apache/spark/sql/SparkSessionExtensionSuite.scala

```diff
@@ -116,6 +121,23 @@ class SparkSessionExtensionSuite extends SparkFunSuite {
     }
   }
 
+  test("inject columnar") {
+    val extensions = create { extensions =>
+      extensions.injectColumnar(session => MyColumarRule(PreRuleReplaceAdd(), MyPostRule()))
+    }
+    withSession(extensions) { session =>
+      assert(session.sessionState.columnarRules.contains(
+        MyColumarRule(PreRuleReplaceAdd(), MyPostRule())))
+      import session.sqlContext.implicits._
+      // repartitioning avoids having the add operation pushed up into the LocalTableScan
+      val data = Seq((100L), (200L), (300L)).toDF("vals").repartition(1)
+      val result = data.selectExpr("vals + 1").collect()
```

Comment:
maybe lets make the rule `PreRuleReplaceAddPlusExtraOne` as I see that 100 + 1 is not 102, but you stated this within the rule :) 

re: https://github.com/apache/spark/pull/24795/files#diff-b7face523211086b889c334ae4075ad2R549